activesupport/lib/active_support/testing/parallelize_when_suitable_executor.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module Testing
+    class ParallelizeWhenSuitableExecutor
+      def initialize(parallel_executor)
+        @parallel_executor = parallel_executor
+      end
+
+      def start
+        parallelize if should_parallelize?
+        show_execution_info
+
+        @parallel_executor.start if @parallelized
+      end
+
+      def <<(work)
+        @parallel_executor << work
+      end
+
+      def shutdown
+        @parallel_executor.shutdown if @parallelized
+      end
+
+      private
+        def parallelize
+          @parallelized = true
+          Minitest::Test.parallelize_me!
+        end
+
+        def should_parallelize?
+          tests_count > ActiveSupport.test_parallelization_minimum_number_of_tests
+        end
+
+        def tests_count
+          Minitest::Runnable.runnables.sum { |runnable| runnable.runnable_methods.size }
+        end
+
+        def show_execution_info
+          puts execution_info
+        end
+
+        def execution_info
+          if should_parallelize?
+            executor_label = @parallel_executor.is_a?(ActiveSupport::Testing::Parallelization) ? "processes" : "threads"
```

Comment:
Instead of doing an `is_a` here what do you think about passing the option along with `initialize`?